lib/modules/apostrophe-schemas/index.js

```diff
@@ -1455,6 +1455,9 @@ module.exports = {
           },
           safeFor: 'manage',
           launder: function(value) {
+            if ((typeof field.choices) === 'string') {
+              field.choices = self.apos.modules[field.moduleName][field.choices](cursor.get('req'), field);
```

Comment:
The dynamic choices function may return a promise (although it is not guaranteed to), so this won't be sufficient. Look at the promise chain used for this purpose in "convert" for select elements.

Unfortunately launder is synchronous. And ultimately this has to work in queryToFilters, which is also synchronous.

This would take some time to architect a complete solution for.

Reply:
Maybe we can simplify. The important thing about these choice functions is to show what the options are. There is no actual harm in allowing someone to filter by the wrong string. They just won't get anything.

So I think that in the case where the choices are dynamic, launder should just use self.apos.launder.string.